lib/src/utils.ts

```diff
@@ -90,8 +90,14 @@ export function valueError(message: string, name?: string): Error {
 export function pathToUrlString(path: string): string {
   if (p.isAbsolute(path)) return url.pathToFileURL(path).toString();
 
-  const components = p.sep === '\\' ? path.split(/[/\\]/) : path.split('/');
-  return components.map(encodeURIComponent).join('/');
+  // percent encode relative path like `pathToFileURL`
+  return encodeURI(path)
+    .replace(/[#?]/g, encodeURIComponent)
+    .replace(
+      process.platform === 'win32' ? /%(5B|5C|5D|5E|7C)/g : /%(5B|5D|5E|7C)/g,
+      decodeURIComponent
+    )
```

Comment:
I think this is supposed to be `encodeURIComponent`... currently this will convert the path `%5B` to the URL `[`, when it should be converted to `%255B`.

Reply:
This is intended. `pathname` of URL has a different [percent-encode set](https://url.spec.whatwg.org/#path-percent-encode-set) that some of the character does not need to be escaped.

Reply:
And it will not encode `%5B` to `[` because first `%` will be encoded as `%25` first, thus there is no `%5B` to be replaced.